web_src/js/modules/tippy.js

```diff
@@ -3,6 +3,11 @@ import tippy from 'tippy.js';
 const visibleInstances = new Set();
 
 export function createTippy(target, opts = {}) {
+  const {role, content, onHide: optsOnHide, onDestroy: optsOnDestroy, onShow: optOnShow} = opts;
+  delete opts.onHide;
+  delete opts.onDestroy;
+  delete opts.onShow;
```

Comment:
Otherwise, the `...opts,` below would overwrite the createTippy's `onXxx`

Reply:
Mutating function args is bad, please avoid in the future. In this case, you could just create a new array with spread syntax:

```js
const {onHide, onShow, onDestroy, ...other} = opts;
```

Now `other` holds all opts besides those three and does not mutate the args.

Reply:
I do not see any problem for doing this in JS. Because in JS nothing is really immutable, the caller should create new objects for each call.

And, introducing `other` makes the code unclear for when to use `opts` or when to use `other`.

But if you prefer, I can take it in #25253 by the way.

Reply:
Yes, the caller should, but in case you have no control over the caller, it's better to not mutate to not intruduce bugs in the caller. The only safe mutations are for primitive types like string, but I'd argue for clarity it's better to not even mutate those.